kubernetes/config/incluster_config.py

```diff
@@ -16,13 +16,15 @@
 
 from kubernetes.client import configuration
 
-_SERVICE_HOST_ENV_NAME = "KUBERNETES_SERVICE_HOST"
-_SERVICE_PORT_ENV_NAME = "KUBERNETES_SERVICE_PORT"
-_SERVICE_TOKEN_FILENAME = "/var/run/secrets/kubernetes.io/serviceaccount/token"
-_SERVICE_CERT_FILENAME = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+from .config_exception import ConfigException
 
+SERVICE_HOST_ENV_NAME = "KUBERNETES_SERVICE_HOST"
+SERVICE_PORT_ENV_NAME = "KUBERNETES_SERVICE_PORT"
+SERVICE_TOKEN_FILENAME = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+SERVICE_CERT_FILENAME = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
 
-def _join_host_port(host, port):
+
+def join_host_port(host, port):
```

Comment:
I agree that tests should not respect that convention, but to be able to use this method outside of this module, one should import it in __init__. Because I saw that gateway exists and it is still private to module, I did not see any problem make it package access. I however have no strong opinion about it, so I will just revert this one, but we can discuss if __init__ gateway is enough for being private inside a package or we need to always follow _ convention.